### PR TITLE
Emoji mart

### PR DESCRIPTION
**Pull request by bago2k4**
Replace our current picker with EmojiPicker.
Card: https://trello.com/c/jRmI3Ta8

To test:
- try to add emojis with the picker in the headline
- try to add emojis with the picker in the body
- try to add emojis with the picker in a comment
- try to add emojis with the autcomplete (this still uses emojione lib) in all the above places

### CODE REVIEW

File: src/oc/web/components/ui/emoji_picker.cljs [hunk 1]
```diff
@@ -115,21 +101,17 @@
...
-                                                           (remove-markers s)
-                                                           (reset! visible false)
-                                                           (.focus @last-active-element)
-                                                           (when (fn? add-emoji-cb)
-                                                              (add-emoji-cb @last-active-element emoji)))}))]]))
\ No newline at end of file
+          (react-utils/build (.-Picker js/EmojiMart)
+           {:native true
+            :onClick (fn [emoji event]
+                       (js/console.log "Picker/onClick" emoji)
```
Comment (thepug) [inline src/oc/web/components/ui/emoji_picker.cljs]: remove or use `timbre`

**Comment by bago2k4** [comment]
@thepug ready again

**Comment by thepug** [comment]
One issue I noticed when editing the post.
http://recordit.co/FM4DGTGEDC
fm4dgtgedc

Maybe just disable the comments while editing the post?  I doubt anyone will do this.

**Comment by bago2k4** [comment]
@thepug the problem was more at the source, before we had only one picker at the time so this wasn't a problem.
I solved passing to the emoji-picker a selector that represents the container where the focusable field should be in to be enabled.
As before the focusable elements have to have the `emojiable` class to enable the picker button.
Ready again for testing.

File: src/oc/web/components/ui/emoji_picker.cljs [hunk 2]
```diff
@@ -28,7 +31,7 @@
 (defn save-caret-position [s]
   (remove-markers s)
   (let [caret-pos (::caret-pos s)]
-    (if (>= (.indexOf (.-className (.-activeElement js/document)) emojiable-class) 0)
+    (if (emojiable-active?)
```
Comment (thepug) [inline src/oc/web/components/ui/emoji_picker.cljs]: ❤️

**Comment by thepug** [comment]
Well, there are still some issues with the emoji picker being disabled.

http://recordit.co/yFBIRBuHRu/
yfbirbuhru

**Comment by bago2k4** [comment]
yeah @thepug i saw that, should be good now..

**Comment by thepug** [comment]
[Emoji-mart and reactions picker](https://trello.com/c/jRmI3Ta8/1098-emoji-mart-and-reactions-picker)

**Pull request merged**